src/satellite_images_nso_datascience/model_metrics/custom_model_metrics.py

```diff
@@ -0,0 +1,296 @@
+            self.satellite_constellation == "PNEO"
+            and self.location == "Nieuwkoopse_plassen"
+        ):
+            collect_files = glob.glob(
+                os.path.abspath("./tests/test_input/").replace("\\", "/")
+                + "/Nieuwkoopse_plassen_raster_images/*PNEO*.tif"
+            )
+
+        # Make a iterator to make predictions with.
+        for a_tif_file in collect_files:
+            a_tif_file = a_tif_file.replace("\\", "/")
+            print(a_tif_file)
+
+            with contextlib.redirect_stdout(io.StringIO()):
+                output_file_name_generator = OutputFileNameGenerator(
+                    output_path=os.path.abspath("./tests/test_output/"),
+                    output_file_name=os.path.abspath("./tests/test_output/")
+                    + a_tif_file.split("/")[-1].replace(".tif", ".parquet"),
+                )
+
+                nso_tif_kernel_iterator_generator = (
+                    tif_model_iterator.TifModelIteratorGenerator(
+                        path_to_tif_file=a_tif_file,
+                        model=self.model,
+                        output_file_name_generator=output_file_name_generator,
+                        parts=1,
+                        normalize_scaler=self.scaler,
+                        column_names=self.selected_features,
+                        dissolve_parts=False,
+                        square_output=False,
+                        do_all_parts=True,
+                    )
+                )
+
+                nso_tif_kernel_iterator_generator.predict_all_output()
+
+        # Collect the .parquet files to see how the predictions are.
+        if (
+            self.satellite_constellation == "Superview"
+            and self.location == "Nieuwkoopse_plassen"
+        ):
+            collect_files = glob.glob(
+                os.path.abspath("./tests/test_output/").replace("\\", "/")
+                + "*SV*.parquet"
+            )
+        elif (
+            self.satellite_constellation == "PNEO"
+            and self.location == "Nieuwkoopse_plassen"
+        ):
+            collect_files = glob.glob(
+                os.path.abspath("./tests/test_output/").replace("\\", "/")
+                + "*PNEO*.parquet"
+            )
+
+        return_metrics = []
+
+        for afile in collect_files:
+            afile = afile.replace("\\", "/")
+
+            # Get the most common prediction file and base that as it's predicted value.
+            predict_value = pd.read_parquet(afile)["label"].value_counts().index[0]
+
+            return_metrics.append(
+                [
+                    afile,
+                    predict_value == afile.split("_test")[0].split("_")[-1],
+                    predict_value,
+                    afile.split("_test")[0].split("_")[-1],
+                ]
+            )
+
+            os.remove(afile)
+
+        return return_metrics
+
+    def __raster_to_dataframe(self, a_tif_file) -> pd.DataFrame:
```

Comment:
Is this not very close to the TifModelIteratorGenerator._create_pixel_coordinate_dataframe method? Is there a way to avoid code duplication here?

Reply:
yes, but we are using this outside of TifModelIteratorGenerator, to avoid a dependencies and faster use of code

Reply:
I don't agree with you, but let's leave it for now.